LogBlockIndexWriterTest#shouldAppendBlockAfterRecoverFromSnapshot is flaky

```
15:33:07 [1 - Java Tests] [ERROR] Tests run: 10, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 14.683 s <<< FAILURE! - in io.zeebe.logstreams.log.LogBlockIndexWriterTest
15:33:07 [1 - Java Tests] [ERROR] shouldAppendBlockAfterRecoverFromSnapshot(io.zeebe.logstreams.log.LogBlockIndexWriterTest)  Time elapsed: 10.107 s  <<< FAILURE!
15:33:07 [1 - Java Tests] org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>
15:33:07 [1 - Java Tests] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
15:33:07 [1 - Java Tests] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
15:33:07 [1 - Java Tests] 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
15:33:07 [1 - Java Tests] 	at io.zeebe.test.util.TestUtil$Invocation.until(TestUtil.java:96)
15:33:07 [1 - Java Tests] 	at io.zeebe.test.util.TestUtil$Invocation.until(TestUtil.java:69)
15:33:07 [1 - Java Tests] 	at io.zeebe.test.util.TestUtil.waitUntil(TestUtil.java:41)
15:33:07 [1 - Java Tests] 	at io.zeebe.logstreams.log.LogBlockIndexWriterTest.shouldAppendBlockAfterRecoverFromSnapshot(LogBlockIndexWriterTest.java:256)
15:33:07 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
15:33:07 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
15:33:07 [1 - Java Tests] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
15:33:07 [1 - Java Tests] 	at java.lang.reflect.Method.invoke(Method.java:498)
15:33:07 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
15:33:07 [1 - Java Tests] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
15:33:07 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
15:33:07 [1 - Java Tests] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
15:33:07 [1 - Java Tests] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
15:33:07 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:33:07 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:33:07 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:33:07 [1 - Java Tests] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
15:33:07 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
15:33:07 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
15:33:07 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:128)
15:33:07 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:27)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
15:33:07 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
15:33:07 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
15:33:07 [1 - Java Tests] 
```

--- a/logstreams/src/test/java/io/zeebe/logstreams/log/LogBlockIndexWriterTest.java
+++ b/logstreams/src/test/java/io/zeebe/logstreams/log/LogBlockIndexWriterTest.java
@@ -252,6 +252,8 @@ public class LogBlockIndexWriterTest {
     logStreamRule.getClock().pinCurrentTime();
     writer.writeEvents(2, EVENT_1, true);
 
+    waitUntil(() -> !blockIndex.isEmpty());
+
     logStreamRule.getClock().addTime(SNAPSHOT_INTERVAL);
     waitUntil(() -> getSnapshotCount() > 0);
 

